android/src/main/java/com/mastercard/compass/cp3/lib/react_native_wrapper/route/WriteProgramSpaceAPIRoute.kt

```diff
@@ -0,0 +1,80 @@
+package com.mastercard.compass.cp3.lib.react_native_wrapper.route
+
+import android.app.Activity
+import android.content.Intent
+import android.util.Log
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReadableMap
+import com.google.gson.Gson
+import com.mastercard.compass.base.Constants
+import com.mastercard.compass.cp3.lib.react_native_wrapper.ui.WriteProgramSpaceCompassApiHandlerActivity
+import com.mastercard.compass.cp3.lib.react_native_wrapper.ui.util.*
+import com.mastercard.compass.cp3.lib.react_native_wrapper.ui.util.Collection
+import com.mastercard.compass.cp3.lib.react_native_wrapper.util.ErrorCode
+import com.mastercard.compass.cp3.lib.react_native_wrapper.util.Key
+import com.mastercard.compass.model.programspace.WriteProgramSpaceDataResponse
+import timber.log.Timber
+import java.util.Date
+
+class WriteProgramSpaceAPIRoute(
+  private val context: ReactApplicationContext,
+  private val currentActivity: Activity?
+) {
+  companion object {
+    val REQUEST_CODE_RANGE = 1000 until 1100
+
+    const val WRITE_PROGRAM_SPACE_REQUEST_CODE = 901
+    const val TAG = "WriteProgramSpaceAPIRoute"
+  }
+
+  fun startWriteProgramSpaceIntent(
+    WriteProgramSpaceParams: ReadableMap
+  ) {
+    val reliantGUID: String = WriteProgramSpaceParams.getString("reliantGUID")!!
+    val programGUID: String = WriteProgramSpaceParams.getString("programGUID")!!
+    val rID: String = WriteProgramSpaceParams.getString("rID")!!
+    val jsonData: String = WriteProgramSpaceParams.getString("programSpaceData")!!
+    val encryptData: Boolean = WriteProgramSpaceParams.getBoolean("encryptData")
+
+    Timber.d("reliantGUID: $reliantGUID")
+    Timber.d("programGUID: $programGUID")
+    Timber.d("rID: $rID")
+    Timber.tag(TAG).d(encryptData.toString())
+
+    val intent = Intent(context, WriteProgramSpaceCompassApiHandlerActivity::class.java).apply {
+      putExtra(Key.RELIANT_APP_GUID, reliantGUID)
+      putExtra(Key.PROGRAM_GUID, programGUID)
+      putExtra(Key.RID, rID)
+      putExtra(Key.PROGRAM_SPACE_DATA, jsonData)
+      putExtra(Key.ENCRYPT_DATA, encryptData)
+    }
+
+    currentActivity?.startActivityForResult(intent, WRITE_PROGRAM_SPACE_REQUEST_CODE)
+  }
+
+  fun handleWriteProgramSpaceIntentResponse(
+    resultCode: Int,
+    data: Intent?,
+    promise: Promise
+  ) {
+    when (resultCode) {
+      Activity.RESULT_OK -> {
+        val resultMap = Arguments.createMap()
+        val response: WriteProgramSpaceDataResponse = data?.extras?.get(Key.DATA) as WriteProgramSpaceDataResponse
+
+        Timber.tag(TAG).d(response.toString())
+        resultMap.putBoolean("isSuccess", response.isSuccess)
```

Comment:
This may resolve as either true or false. If it resolves as false we need to throw an error back with the code in the Response object.

Reply:
Resolved [here](https://github.com/Mastercard/community-pass-react-native-wrapper/pull/17/commits/35b625b1fc64c7eaed4282b21819895eedb5bb21)